### DISCUSSION TITLE
Meaning of index parameter in hook function signature

### DISCUSSION THREAD

**Discussion by mattrossman** [pmndrs/use-cannon#121] [Q&A] [answered]
Most of the hooks accept some function (e.g. `PlaneFn`, `BoxFn` etc.)
Each of these has a format like `(index: number) => SomeProps`.

What is the meaning of `index` in that function signature? I don't see this explained or used in any examples.

**Comment by stockhuman** [comment, marked as answer]
Better late than never?

You'll find your answer on [L183](https://github.com/pmndrs/use-cannon/blob/576d7967935dbbfcd44b81347caab43487382702/src/hooks.ts#L183) of _hooks.ts_:
```js
props = uuid.map((id, i) => {
  const props = prepare(temp, fn(i), argFn) // here, fn is the x-Fn you mentioned
  temp.updateMatrix()
  object.setMatrixAt(i, temp.matrix)
  object.instanceMatrix.needsUpdate = true
  return props
})
```

It's a UUID index, used to iterate over every body in the scene.
